testdata/sqllogictests/create_table.slt

```diff
@@ -9,6 +9,9 @@ set search_path to create_table_schema;
 statement ok
 create table t1 (a int, b int);
 
+statement ok
+create or replace table t1 (a int, b int);
+
```

Comment:
Could we add a test that adds a new column for the replace? The scenario I want to see tested is:
- Create a table with 2 columns
- Replace with a table with 3 columns
- Do inserts with 3 columns work?
- What happens to existing table data?

Reply:
hmm, yeah definitely a few things need to happen in the delta writer too. Currently this just orphans the files, but we'd instead want to update the existing delta table using delta commands.